src/network.js

```diff
@@ -536,23 +540,37 @@ Network.prototype = {
     if (!this.optimized)
       this.optimize();
 
-    var hardcode = "var inputs = " + this.optimized.data.inputs.length + ";\n";
-    hardcode += "var outputs = " + this.optimized.data.outputs.length + ";\n";
-    hardcode += "var F =  new Float64Array([" + this.optimized.memory.toString() + "]);\n";
-    hardcode += "var activate = " + this.optimized.activate.toString() + ";\n";
-    hardcode += "var propagate = " + this.optimized.propagate.toString() + ";\n";
-    hardcode +=
-        "onmessage = function(e) {\n" +
-          "if (e.data.action == 'startTraining') {\n" +
-            "train(" + JSON.stringify(set) + "," + JSON.stringify(workerOptions) + ");\n" +
-          "}\n" +
-        "}";
-
-    var workerSourceCode = workerFunction + '\n' + hardcode;
-    var blob = new Blob([workerSourceCode]);
-    var blobURL = window.URL.createObjectURL(blob);
-
-    return new Worker(blobURL);
+    var hardcode =
+      "var F = new Float64Array([" + this.optimized.memory.join(',') + "]);\n" +
+      "var network = { " + workerFunction + "\n" +
+      "activate: " + this.optimized.activate.toString() + ",\n" +
+      "propagate: " + this.optimized.propagate.toString() + "}\n" +
+      "var onmessage = function(e) {\n" +
+      "  if (e.data.action == 'startTraining') {\n" +
+      "    network.train(" + JSON.stringify(set) + "," + JSON.stringify(workerOptions) + ");\n" +
+      "  }\n" +
+      "}";
+
+    if (typeof(window)!='undefined' && window.Blob && window.Worker) {
+      var blob = new Blob([hardcode]);
+      var blobURL = window.URL.createObjectURL(blob);
+
+      return new Worker(blobURL);
+    }
+    else if (require('child_process')) {
+      // Mimic WebWorker in Node.js environment.
+      var cProcess = require('child_process');
+      var proc = cProcess.fork(__dirname+'/node-worker.js', [hardcode]);
+      proc.postMessage = function(msg){ proc.send(msg) };
+      proc.onmessage = function(){ console.error('must be implemented by the client') };
```

Comment:
for node.js env console output is not very informative, possibly some context mention will be useful